src/core/layout/qgslayoutitemlabel.cpp

```diff
@@ -123,6 +124,11 @@ void QgsLayoutItemLabel::draw( QgsLayoutItemRenderContext &context )
   {
     rectScale = context.renderContext().scaleFactor();
   }
+  else if ( mMode == QgsLayoutItemLabel::ModeHtml && mHtmlUsesTextDocument )
+  {
+    rectScale = context.renderContext().scaleFactor() * 3.67;
+    painter->scale( context.renderContext().scaleFactor() / 3.67, context.renderContext().scaleFactor() / 3.67 );
```

Comment:
Where's the 3.67 value come from?

Reply:
@nyalldawson , yours truly has no answer here; the value has been identified through trial and error, iterating until a perfect match between that and the non-HTML rendering was achieved. 

I suppose it's some unit conversion between QgsTextDocument and our rendering context. I've made a note about the mysterious (and valid) nature of the value.